### PR TITLE
Moved code from documentation to script.

### PR DESCRIPTION
**Pull request by mm-s**
The file docs/BUILD-manual describes the process for executing clone->build->rebuild->....
I found it improveable, after it would be a hassle for me to go capure the script from the README file for performing building cycles in a continuous manner.

I created scripts/make.sh automatizing the tasks described in docs/BUILD-manual.
macOs users invoke CMake differently. The program handles. (it should work on macOS - It is not tested though)

tested on: linux/bash 5.1
Need tests on: MacOS

It doesn't alter the current infrastructure, nothing can break foreseeable.

Note: It is possible that the zsh dependency be no longer needed, I haven't removed it because I cannot evaluate the risk.
The process has been tested successfully in bash 5.1 (the default shell in debian/unstable) successfully.

(Supersedes PR https://github.com/nemtech/catapult-server/pull/145)

### CODE REVIEW

**Comment by mm-s** [comment]
an alternative potential name for this script would be scripts/configure.sh

File: scripts/make.sh [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+Note: If you want the CAT_DEPS_DIR environment variable to persist across sessions make sure to include the last line in the ~/.profile or ~/.bashrc files.
+EOF
+	fi
+	exit 0
+}
+
+function reqroot {
+	if [ "_`whoami`" != "_root" ]; then
+	  echo "Please run as root. (or use sudo)"
+	  exit 1
```
Comment (Jaguar0625) [inline scripts/make.sh:46]: indent with tabs

File: scripts/make.sh [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+function build_catapult {
+	set_depsdir
+	
+	echo "building using ${jobs} jobs"
+	echo "dependencies dir: ${depsdir}"
+        if [ ! -d ${boost_output_dir} ]; then
+                install_deps
+        fi
```
Comment (Jaguar0625) [inline scripts/make.sh:249]: check whitespace

File: scripts/make.sh [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	set +e
+	exitok
+}
+
+#-------------------------------------------------------
+
+prog=$0
+jobs=8
+warn_env=0
+debs="git gcc g++ cmake curl libssl-dev ninja-build zsh pkg-config"
```
Comment (Jaguar0625) [inline scripts/make.sh:275]: debs or deps ?

Comment (Jaguar0625) [reply scripts/make.sh:275]: is zsh still required?

File: scripts/make.sh [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+	build_catapult $@
+fi
+
+#error flow
+help
+exit 1
+
+
+
+
```
Comment (Jaguar0625) [inline scripts/make.sh:294]: delete blank lines

**Review by gimer — COMMENTED** [review]
Can we name it something like `manual-make.sh` in case we'd want to have similar `conan-make.sh` + `conan-make.bat` ?

File: scripts/make.sh (hunk 3 above)
Comment (mm-s) [reply scripts/make.sh:275]: debs, (debian packages).
I haven't found zsh necessary for completing BUILD-manual. I don't know at wider scope.

**Comment by mm-s** [comment]
> Can we name it something like `manual-make.sh` in case we'd want to have similar `conan-make.sh` + `conan-make.bat` ?

I'd suggest extending functionality using more sub-commands and modifiers, e.g.
make.sh --conan
Renaming to manual-make.sh is not a problem though, let me know what's the preferred way.

File: scripts/make.sh (hunk 1 above)
Comment (mm-s) [reply scripts/make.sh]: done

File: scripts/make.sh (hunk 2 above)
Comment (mm-s) [reply scripts/make.sh]: done

File: scripts/make.sh (hunk 3 above)
Comment (mm-s) [reply scripts/make.sh]: I've added libpython-dev to debs list.

**Comment by mm-s** [comment]
I changed the proposed name to configure.sh, instead of make.sh.
make.sh might result more confusing, since this step is not meant to compile sources but rather creating a ready-to-compile _build directory.

please comment on the name
A) scripts/configure.sh --conan
B) scripts/configure-manual.sh + scripts/configure-conan.sh

File: docs/BUILD-manual.md [hunk 5]
```diff
@@ -16,190 +16,53 @@ Required:
...
 ```
 
-## Step 2: Build and install all dependencies
+> **NOTE**:
+> If you only want to download the dependencies (without building and installing them) enter the command ``scripts/configure.sh download deps`` instead. Type ``scripts/configure.sh --help`` for its summary of commands.
+
+## Step 2: Prepare build directory
 
-Copy & paste the whole snippet below into a terminal:
+For creating/updating the _build directory type into a terminal:
```
Comment (Jaguar0625) [inline docs/BUILD-manual.md:49]: please quote ``_build`` because _ is used in markdown formatting

**Comment by Jaguar0625** [comment]
zsh was only used for BUILD-manual, if not needed anymore, you can remove it

File: docs/BUILD-manual.md (hunk 5 above)
Comment (mm-s) [reply docs/BUILD-manual.md]: done

**Review by Jaguar0625 — APPROVED** [review]

**Comment by gimer** [comment]
> please comment on the name
> A) scripts/configure.sh --conan
> B) scripts/configure-manual.sh + scripts/configure-conan.sh

imo b) as in case of windows, there probably won't be "non-conan" script, IDK

**Comment by mm-s** [comment]
> > please comment on the name
> > A) scripts/configure.sh --conan
> > B) scripts/configure-manual.sh + scripts/configure-conan.sh
> 
> imo b) as in case of windows, there probably won't be "non-conan" script, IDK

done.

**Review by gimer — APPROVED** [review]

**Comment by Jaguar0625** [comment]
Merged as 7fd0e3166.

**Pull request closed without merging**